src/config.py

```diff
@@ -35,48 +32,46 @@ def get_env_variable(var_name: str,
     ------
     KeyError
         If allow_empty is False and the environment variable is empty string or None
-    ValueError
-        If cast_to is not compatible with the value stored.
     """
-    env_var = os.getenv(var_name, default)
+    env_var = os.getenv(var_name)
+    if default is not None and env_var in (None, ""):
+        # Set env_var to default, but do not override empty str with None
+        env_var = default
     if not allow_empty and env_var in (None, ""):
         raise KeyError(f"Environment variable {var_name} not set, and allow_empty is False")
-    if isinstance(env_var, cast_to):
-        return env_var
-    # noinspection PyTypeChecker
-    return _cast_str(env_var, cast_to)
+    return env_var
 
 
-def _cast_str(str_to_cast: str, cast_to: Type[T]) -> T:
+def get_bool_env_variable(var_name: str, default: Optional[bool] = None, allow_empty: bool = False) -> bool:
     """
-    Takes a string and casts it to necessary primitive builtin types. Tested with int, float, and bool.
-    For bool, this detects if the value is in the case-insensitive sets {"True", "T", "1"} or {"False", "F", "0"}
-    and raises a ValueError if not. For example _cast_str("False", bool) -> False
+    Reads an environment variable and attempts to cast to bool, with settings to allow defaults, empty values.
+    For bool we have the problem where bool("False") == True but we want this function to return False
 
     Parameters
     ----------
-    str_to_cast : str
-        The string that is going to be casted to the type
-    cast_to : Type[T]
-        The type to cast to e.g. bool
+    var_name : str
+        The name of the environment variable to retrieve.
+    default : Optional[bool] = None
+        Default return value if the environment variable does not exist.
+    allow_empty : bool
+        If False then a KeyError will be raised if the environment variable is empty.
 
     Returns
     -------
-    The string casted to type T defined by cast_to.
+    bool
+        The environment variable, or default if it does not exist
 
     Raises
     ------
-    ValueError if [cast_to] is not compatible with the value stored.
+    ValueError
+        If allow_empty is False and the environment variable is empty string or None
     """
-    # Special cases i.e. casts that aren't of the form int("7") -> 7
-    if cast_to == bool:
-        # For bool we have the problem where bool("False") == True but we want this function to return False
-        truth_values = {"true", "t", "1"}
-        false_values = {"false", "f", "0", ""}
-        if str_to_cast is None or str_to_cast.lower() in false_values:
-            return False
-        elif str_to_cast.lower() in truth_values:
-            return True
-        raise ValueError(f"{str_to_cast} being casted to bool but is not in {truth_values} or {false_values}")
-    # General case
-    return cast_to(str_to_cast)
+    env_variable = get_env_variable(var_name, default, allow_empty)
```

Comment:
```suggestion
    env_variable = get_env_variable(var_name, str(default), allow_empty)
```

This change is from the NewZealidar link I had in the issue, and should fix your failing test